Compiler errors with libc++ 13.0.0

```
clang++: scipy/interpolate/_rbfinterp_pythran.cpp
In file included from scipy/interpolate/_rbfinterp_pythran.cpp:1:
In file included from /usr/lib/python3.9/site-packages/pythran/pythonic/core.hpp:36:
In file included from /usr/lib/python3.9/site-packages/pythran/pythonic/types/slice.hpp:5:
In file included from /usr/lib/python3.9/site-packages/pythran/pythonic/types/NoneType.hpp:8:
In file included from /usr/lib/python3.9/site-packages/pythran/pythonic/builtins/bool_.hpp:7:
In file included from /usr/lib/python3.9/site-packages/pythran/pythonic/types/tuple.hpp:13:
In file included from /usr/lib/python3.9/site-packages/pythran/pythonic/types/ndarray.hpp:10:
In file included from /usr/lib/python3.9/site-packages/pythran/pythonic/builtins/ValueError.hpp:6:
In file included from /usr/lib/python3.9/site-packages/pythran/pythonic/types/exceptions.hpp:6:
/usr/lib/python3.9/site-packages/pythran/pythonic/types/str.hpp:407:12: error: no matching constructor for initialization of 'typename str::reverse_iterator' (aka 'reverse_iterator<(anonymous namespace)::pythonic::types::string_iterator>')
    return {data->rbegin()};
           ^~~~~~~~~~~~~~~~
/usr/include/c++/v1/__iterator/reverse_iterator.h:37:28: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'reverse_iterator<std::basic_string<char>::iterator>' to 'const reverse_iterator<(anonymous namespace)::pythonic::types::string_iterator>' for 1st argument
class _LIBCPP_TEMPLATE_VIS reverse_iterator
                           ^
/usr/include/c++/v1/__iterator/reverse_iterator.h:37:28: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'reverse_iterator<std::basic_string<char>::iterator>' to 'reverse_iterator<(anonymous namespace)::pythonic::types::string_iterator>' for 1st argument
/usr/include/c++/v1/__iterator/reverse_iterator.h:79:14: note: candidate constructor not viable: no known conversion from 'std::basic_string<char>::reverse_iterator' (aka 'reverse_iterator<__wrap_iter<char *>>') to '(anonymous namespace)::pythonic::types::string_iterator' for 1st argument
    explicit reverse_iterator(_Iter __x) : __t(__x), current(__x) {}
             ^
/usr/include/c++/v1/__iterator/reverse_iterator.h:85:5: note: candidate template ignored: substitution failure [with _Up = std::__wrap_iter<char *>]: no member named '_EnableIfImpl' in 'std::_MetaBase<false>'
    reverse_iterator(const reverse_iterator<_Up>& __u)
    ^
/usr/include/c++/v1/__iterator/reverse_iterator.h:76:5: note: candidate constructor not viable: requires 0 arguments, but 1 was provided
    reverse_iterator() : __t(), current() {}
    ^
In file included from scipy/interpolate/_rbfinterp_pythran.cpp:1:
In file included from /usr/lib/python3.9/site-packages/pythran/pythonic/core.hpp:36:
In file included from /usr/lib/python3.9/site-packages/pythran/pythonic/types/slice.hpp:5:
In file included from /usr/lib/python3.9/site-packages/pythran/pythonic/types/NoneType.hpp:8:
In file included from /usr/lib/python3.9/site-packages/pythran/pythonic/builtins/bool_.hpp:7:
In file included from /usr/lib/python3.9/site-packages/pythran/pythonic/types/tuple.hpp:13:
In file included from /usr/lib/python3.9/site-packages/pythran/pythonic/types/ndarray.hpp:10:
In file included from /usr/lib/python3.9/site-packages/pythran/pythonic/builtins/ValueError.hpp:6:
In file included from /usr/lib/python3.9/site-packages/pythran/pythonic/types/exceptions.hpp:6:
/usr/lib/python3.9/site-packages/pythran/pythonic/types/str.hpp:417:12: error: no matching constructor for initialization of 'typename str::reverse_iterator' (aka 'reverse_iterator<(anonymous namespace)::pythonic::types::string_iterator>')
    return {data->rend()};
           ^~~~~~~~~~~~~~
/usr/include/c++/v1/__iterator/reverse_iterator.h:37:28: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'reverse_iterator<std::basic_string<char>::iterator>' to 'const reverse_iterator<(anonymous namespace)::pythonic::types::string_iterator>' for 1st argument
class _LIBCPP_TEMPLATE_VIS reverse_iterator
                           ^
/usr/include/c++/v1/__iterator/reverse_iterator.h:37:28: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'reverse_iterator<std::basic_string<char>::iterator>' to 'reverse_iterator<(anonymous namespace)::pythonic::types::string_iterator>' for 1st argument
/usr/include/c++/v1/__iterator/reverse_iterator.h:79:14: note: candidate constructor not viable: no known conversion from 'std::basic_string<char>::reverse_iterator' (aka 'reverse_iterator<__wrap_iter<char *>>') to '(anonymous namespace)::pythonic::types::string_iterator' for 1st argument
    explicit reverse_iterator(_Iter __x) : __t(__x), current(__x) {}
             ^
/usr/include/c++/v1/__iterator/reverse_iterator.h:85:5: note: candidate template ignored: substitution failure [with _Up = std::__wrap_iter<char *>]: no member named '_EnableIfImpl' in 'std::_MetaBase<false>'
    reverse_iterator(const reverse_iterator<_Up>& __u)
    ^
/usr/include/c++/v1/__iterator/reverse_iterator.h:76:5: note: candidate constructor not viable: requires 0 arguments, but 1 was provided
    reverse_iterator() : __t(), current() {}
    ^
scipy/interpolate/_rbfinterp_pythran.cpp:2291:17: warning: unknown attribute 'externally_visible' ignored [-Wunknown-attributes]
__attribute__ ((externally_visible))
                ^~~~~~~~~~~~~~~~~~
1 warning and 2 errors generated.
```

the issue is the following, for emphasis:

```
/usr/lib/python3.9/site-packages/pythran/pythonic/types/str.hpp:407:12: error: no matching constructor for initialization of 'typename str::reverse_iterator' (aka 'reverse_iterator<(anonymous namespace)::pythonic::types::string_iterator>')
    return {data->rbegin()};
```

refer to https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=259967 for a similar bug. apparently something changed with LLVM's c++ std library implementation

specifically https://reviews.llvm.org/D81414

--- a/pythran/pythonic/types/str.hpp
+++ b/pythran/pythonic/types/str.hpp
@@ -404,7 +404,8 @@ namespace types
 
   typename str::reverse_iterator str::rbegin() const
   {
-    return {data->rbegin()};
+    string_iterator iter(data->end());
+    return typename str::reverse_iterator(iter);
   }
 
   typename str::iterator str::end() const
@@ -414,7 +415,8 @@ namespace types
 
   typename str::reverse_iterator str::rend() const
   {
-    return {data->rend()};
+    string_iterator iter(data->begin());
+    return typename str::reverse_iterator(iter);
   }
 
   auto str::c_str() const -> decltype(data->c_str())

